diam/reflect.go

```diff
@@ -85,6 +85,16 @@ func marshalStruct(m *Message, field reflect.Value) (error, []*AVP) {
 	for n := 0; n < base.NumField(); n++ {
 		f := base.Field(n)
 		bt := base.Type().Field(n)
+
+		if bt.Anonymous {
+			err, embeddedAvps := marshalStruct(m, f)
+			if err != nil {
+				return err, nil
+			}
+			avps = append(avps, embeddedAvps...)
+			continue
+		}
+
```

Comment:
It will fail if there is non struct embedded field. Something like this could work:

```
		if bt.Anonymous && bt.Type.Kind() == reflect.Struct && len(bt.Tag) == 0 {
			err, embeddedAvps := marshalStruct(m, f)
			if err != nil {
				return err, nil
			}
			avps = append(avps, embeddedAvps...)
			continue
		}

```

Reply:
Fixed in 061a6ec780e303403aa849cf71e3dc3762258b5d